packages/data/src/settings/with-settings-hydration.js

```diff
@@ -0,0 +1,44 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { useRef } from '@wordpress/element';
+import { getAllSettings } from '@woocommerce/wc-admin-settings';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from './constants';
+
+export const withSettingsHydration = group => OriginalComponent => {
+	return props => {
+		const settings = useRef( getAllSettings() );
```

Comment:
I'm sorry I didn't catch this earlier, but I just realized that you are coupling the settings source (`getAllSettings()` from an alias) with an exported package.  

I think what would be better here is if the HOC receives both `group` and `settings` as it's arguments. That way any consumers (which in this case will be the wc-admin _client_) would just provide the settings object as the source for hydrating the internal data store. 

Either that, or:

- Add back in a resolver for the settings api endpoint to the data store.
- use `window.wcSettings` directly here as the source (instead of `getAllSettings`) and if it's undefined, you can allow the hydration to fallback to the resolver.

With all the work that's already gone into this pull, it's probably easier to just do the former option (provide `group` and `settings` as arguments to the `withSettingsHydration` hoc.)